src/object/cli_obj.c

```diff
@@ -2173,6 +2178,10 @@ obj_comp_cb(tse_task_t *task, void *data)
 	case DAOS_OBJ_RPC_QUERY_KEY:
 	case DAOS_OBJ_RPC_SYNC:
 		obj = *((struct dc_object **)data);
+		if (obj_auxi->opc == DAOS_OBJ_RPC_UPDATE ||
+		    obj_auxi->opc == DAOS_OBJ_RPC_FETCH)
+			/** checksums sent, need to destroy now */
+			obj_rw_csum_destroy(obj, dc_task_get_args(task));
```

Comment:
not sure why you move it to here?
the checksum calculate need not to do it again for io retry case right? then only need to destroy csum when need not retry IO, or maybe I misunderstood it?